parla/task_runtime.py

```diff
@@ -853,15 +760,39 @@ class InvalidSchedulerAccessException(RuntimeError):
 
 
 class SchedulerContext(metaclass=ABCMeta):
-    def spawn_task(self, function, args, deps, taskid,
-                   req, dataflow, name: Optional[str] = None):
-        return ComputeTask(function, args, deps, taskid, req, dataflow, name)
-
-    def create_wait_task(self, function, args, deps, taskid,
-                         req, dataflow, num_unspawned_deps,
-                         name: Optional[str] = None):
-        return ComputeTask(function, args, deps, taskid, req,
-                           dataflow, name, num_unspawned_deps)
+    def spawn_task(
+        self,
+        function,
+        args,
+        deps: List[Union[TaskID, Task]],
+        taskid,
+        req,
+        dataflow,
+        name: Optional[str] = None,
+    ):
+        # _flat_tasks appends two types of objects to deps.
+        # If a task corresponding to a task id listed on the dependencies
+        # is already spawned (materialized), it appends the task object.
+        # Otherwise, a task corresponding to the task id is not yet spawned
+        # and, in this case, appends its id which is not spawned yet as a key,
+        # and the dependee task id which waits for the dependent task to the
+        # wait_dependees_collection dictionary wrapper.
+        # The tasks on that dictionary is not spawned until all
+        # dependent tasks are spawned.
```

Comment:
This comment needs updating. `_flat_tasks` is in a different file so it's confusing; also `wait_dependees_collection` no longer exists in this PR.